community/kernel/src/main/java/org/neo4j/kernel/impl/store/id/BufferingIdGeneratorFactory.java

```diff
@@ -32,8 +35,7 @@
  */
 public class BufferingIdGeneratorFactory extends IdGeneratorFactory.Delegate
 {
-    private final BufferingIdGenerator[/*IdType#ordinal as key*/] overriddenIdGenerators =
-            new BufferingIdGenerator[IdType.values().length];
+    private final Map<IdType, BufferingIdGenerator> overriddenIdGenerators = new HashMap<>();
```

Comment:
I'd rather have the array, HashMap is way more expensive, it's an unnecessary cost. Can we introduce a typeId() on IdType instead?